File: code/js/src/Dto/output/channels/ChannelMemberOutputModel.ts
import { ChannelRole } from '../../../Domain/channel/ChannelRole';

/**
 * Output model for channel member
 *
 * @param id The unique identifier of the member.
 * @param name The name of the member.
 * @param role The role of the member in the channel.
 */
export interface ChannelMemberOutputModel {
    id: number;
    name: string;
    role: ChannelRole;
}

File: code/server/jvm/im-http-api/src/main/kotlin/im/api/model/output/channel/ChannelMemberOutputModel.kt
package im.api.model.output.channel

import im.domain.channel.ChannelRole
import im.domain.user.User

data class ChannelMemberOutputModel(
    val id: Long,
    val name: String,
    val role: String,
) {
    companion object {
        fun fromDomain(
            user: User,
            role: ChannelRole,
        ) = ChannelMemberOutputModel(
            id = user.id.value,
            name = user.name.value,
            role = role.name,
        )
    }
}
